### ISSUE TITLE
After some time the choose elem doesnt work anymore and gives out a blueprint and a crash

### ISSUE THREAD

**Issue by LuziferSenpai** [vsaulue/factorio-dana#1]
Hey,

like the title says after 2 runs of the mod with pressing new query it malfunctions. I cant select a item anymore and with a right click I got this bug:
```
The mod Dana: crafting tree/graph explorer (0.1.0) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event dana::on_gui_elem_changed (ID 67)
__dana__/lua/logger/Logger.lua:62: [dana][Error] Invalid access at index: nil
stack traceback:
	[C]: in function 'error'
	__dana__/lua/logger/Logger.lua:62: in function 'error'
	__dana__/lua/containers/ErrorOnInvalidRead.lua:30: in function '__index'
	__dana__/lua/apps/query/gui/IntermediateSetEditor.lua:122: in function 'onElemChanged'
	__dana__/lua/gui/GuiElement.lua:137: in function <__dana__/lua/gui/GuiElement.lua:134>
```

**Comment by vsaulue** [comment]
Hi,

> After some time the choose elem doesnt work anymore and gives out a blueprint

Did you still had the "dana-select-tool" under your cursor (which happens to use the blueprint's icon) ? Is this problem solved if you clean your cursor ?
Factorio's "choose-elem-button" will not open their GUI and "select" whatever you have in the cursor if it's not empty (can be seen as a drag & drop feature). So it's not a bug, but that might not be a desirable/intuitive feature either...

> with a right click I got this bug

These button definitely crash the mod when right clicked, thanks for the catch ! This should be fixed for the next release (which might take some time, as I have to learn about migration scripts before being able to release a v0.1.1 ...)

**Comment by LuziferSenpai** [comment]
No, I even have the full screen disabled. I wanted to select a item and it didnt allowed me to.

**Comment by vsaulue** [comment]
Hi,
fix for the right click issue has been published in v0.1.1

For the button refusing to select & giving a blueprint, We might have a misunderstanding here, as I don't see how being in windowed/full screen is relevant. I was asking if you had a blueprint-type item selected in game. If that's the case, just clearing your in-game cursor (pressint Q in EN/US keyboards) should make the button work properly again.

**Comment by LuziferSenpai** [comment]
No, you dont understand. I dont have ANY gui. So I cant even get a blueprint into my cursor. But yea.

**Comment by vsaulue** [comment]
Could you upload a savefile, with simple steps to reproduce ?
